superset/commands/database/sync_permissions.py

```diff
@@ -0,0 +1,344 @@
+                self.db_connection.name,
+                catalog,
+            )
+            new_catalog_vm = add_vm(db.session, security_manager, new_catalog_perm_name)
+            perm = security_manager.get_catalog_perm(
+                self.old_db_connection_name,
+                catalog,
+            )
+            existing_pvm = security_manager.find_permission_view_menu(
+                "catalog_access",
+                perm,
+            )
+            if existing_pvm:
+                existing_pvm.view_menu = new_catalog_vm
+
+        for schema in schemas:
+            new_schema_perm_name = security_manager.get_schema_perm(
+                self.db_connection.name,
+                catalog,
+                schema,
+            )
+
+            # rename existing schema permission
+            perm = security_manager.get_schema_perm(
+                self.old_db_connection_name,
+                catalog,
+                schema,
+            )
+            existing_pvm = security_manager.find_permission_view_menu(
+                "schema_access",
+                perm,
+            )
+            if existing_pvm:
+                existing_pvm.view_menu.name = new_schema_perm_name
+
+            # rename permissions on datasets and charts
+            for dataset in DatabaseDAO.get_datasets(
+                self.db_connection_id,
+                catalog=catalog,
+                schema=schema,
+            ):
+                dataset.catalog_perm = new_catalog_perm_name
+                dataset.schema_perm = new_schema_perm_name
+                for chart in DatasetDAO.get_related_objects(dataset.id)["charts"]:
+                    chart.catalog_perm = new_catalog_perm_name
+                    chart.schema_perm = new_schema_perm_name
+
+
+@celery_app.task(name="sync_database_permissions", soft_time_limit=600)
+def sync_database_permissions_task(
+    database_id: int, username: str, old_db_connection_name: str
+) -> None:
+    """
+    Celery task that triggers the SyncPermissionsCommand in async mode.
+    """
+    with current_app.test_request_context():
+        try:
+            user = security_manager.get_user_by_username(username)
+            if not user:
+                raise UserNotFoundInSessionError()
+            g.user = user
+            logger.info(
+                "Syncing permissions for DB connection %s while impersonating user %s",
+                database_id,
+                user.id,
+            )
+
+            db_connection = DatabaseDAO.find_by_id(database_id)
+            if not db_connection:
+                raise DatabaseNotFoundError()
+            ssh_tunnel = DatabaseDAO.get_ssh_tunnel(database_id)
+
+            SyncPermissionsCommand(
+                database_id,
+                username,
+                old_db_connection_name=old_db_connection_name,
+                db_connection=db_connection,
+                ssh_tunnel=ssh_tunnel,
+            ).sync_database_permissions()
```

Comment:
The fact that we're calling a specific method of the command instead of `.run()` makes me think that this should've been a function instead of a method (since technically we always want to call only `.run()` when interacting with a command). But this is fine for now, it's just a nit.